### ISSUE TITLE
Empty attributes in inline SVG get stripped

### ISSUE THREAD

**Issue by andreashaerter** [tdewolff/minify#576]
Hello,

empty attributes in inline SVG images are getting stripped (unexpected) while being kept (as expected) in other HTML tags. Example code:

```html
<!DOCTYPE html>
<html lang="en"
      dir="ltr"
      class="no-js">
<head>
  <title>foo</title>
<body x-cloak x-ignore>
  <svg x-ignore x-cloak>foo</a>
  <svg x-ignore x-cloak x-foo="">bar</svg>
  <svg x-ignore x-cloak x-foo="x-foo">barbar</svg>
  <strong x-ignore x-cloak x-foo="">baz</strong>
</body>
</html>
```
results in (line breaks added for better readability)

```html
<!doctype html>
<html lang=en
      dir=ltr
      class=no-js>
<title>foo</title>
<body x-cloak
      x-ignore>
<a x-ignore
   x-cloak>foo</a>
<svg>bar</svg>
<svg x-foo="x-foo">barbar</svg>
<strong x-ignore
        x-cloak
        x-foo>baz</strong>
```
As one can see, empty attributes like `x-ignore x-cloak x-foo=""` are getting stripped from the inline `<svg>` image. I guess this happens as they are handled by the SVG minifier instead of the HTML one (which makes sense but an option to keep empty attributes on SVG would be nice). 

I got this problem while using [Minify as included in the static site generator Hugo](https://gohugo.io/getting-started/configuration/#configure-minify), the issue is reproducible with the [minify online demo](https://go.tacodewolff.nl/minify).

**If somebody else ran into this problem: A workaround is to use the [empty attribute syntax](https://html.spec.whatwg.org/multipage/syntax.html#attributes-2)**

> Just the attribute name. The value is implicitly the empty string.

So e.g. `x-ignore="x-ignore"` in my  example code. The attributes will be preserved then. I ran into this problem as I use [Apline.js's `x-ignore` directive](https://alpinejs.dev/directives/ignore) for the SVG tags to optimize Alpine.js-Performance

**Comment by tdewolff** [comment]
Fixed! Are you sure about the empty attribute syntax? I think it says that you should just write the name (without the `=` and the value), and not that the value should be the name.

**Comment by andreashaerter** [comment]
@tdewolff oh wow, that was fast. Thank you very much.

Yes, I am sure. My brain mixed up [old HTML information](https://www.w3.org/TR/REC-html40/intro/sgmltut.html#h-3.3.4.2) with the formerly linked [empty-attribute-syntax]. However, the information is [still valid for current HTML versions](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes), I edited my original posting.

TL;DR: boolean attributes may legally take a single value: the name of the attribute itself (e.g., selected="selected").

**Comment by vilunov** [comment]
@tdewolff just writing the name is still the wrong way it seems. Firefox for example does not render SVGs with `style` attribute, which is the minified version of `style=""`.

**Comment by tdewolff** [comment]
Thanks, should be fixed now. XML and thus SVG must explicitly have an empty value.
